### DISCUSSION TITLE
[immich] Proper way to backup?

### DISCUSSION THREAD

**Discussion by sebj54** [alexbelgium/hassio-addons#1525] [Q&A]
Hi there!

I installed Immich thanks to your addon. Now that all my photos are stored in Immich, I'd like to back them up. There is [an official guide](https://immich.app/docs/administration/backup-and-restore/) for that, but my attempts are not successful.

I tried to use the console on the Immich container to run the command below, but with an error message:

```
root@db21ed7f-immich:/# pg_dumpall --clean --if-exists --username=postgres | gzip > "/tmp/db.sql"
pg_dumpall: error: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: No such file or directory
        Is the server running locally and accepting connections on that socket?
```

I saw in some issues on this repo that there is already a way to backup the addon itself. How does it work and will it save the Immich database and the files?

Thansk for your help :)

**Comment by alexbelgium** [comment]
Hi, here is how I do it : https://github.com/alexbelgium/hassio-addons/wiki/Performing-backups

**Comment by sebj54** [comment]
Thank you for your quick answer :) 

I just gave it a look but unfortunately, a partial backup with Immich only does not include database or files. So I still have to figure out how to perform a backup manually within the container.

**Reply by alexbelgium** [reply]
So backup of postgres is perform by backup of the postgres add-on, and backup of the immich add-ons backups it's /data. Backup of /config is performed by backup of ha itself. If files are stored on a external drive indeed they are not backuped.

What I do is weekly backups of HA and key add-ons with the addon mentioned in the wiki (Google drive backup), then a copy of my usb drive to a secondary drive by script

**Reply by sebj54** [reply]
Thank you for all the details, it makes sense!

I'll try to make the backup as short as possible by tuning the settings, because actually a complete backup is around 100GB 😅

I'll come back later to this thread to share my experience, it might help someone else :)

**Comment by igi01** [comment]
Is there also a possibility to just automatically backup images to external drive?

If i want to restore a 100gb+ backup with HA wizzard it will take decades.

Thanks :)

**Reply by alexbelgium** [reply]
Ahah, indeed... I think the best is to keep all files in an external drive that is backuped, and use HA just to backup the database?

**Reply by igi01** [reply]
I have a 2 tb internal SSD. 
But will use https://github.com/Poeschl-HomeAssistant-Addons/rsync-local to sync share folder with external SSD

And the Google backup to Backup everything else to the cloud👍

**Reply by alexbelgium** [reply]
Nice!
